crates/services/shared-sequencer/src/http_api.rs

```diff
@@ -0,0 +1,181 @@
+use base64::prelude::*;
+use cosmrs::AccountId;
+
+mod api_types {
+    use serde::Deserialize;
+
+    #[derive(Debug, Deserialize)]
+    pub struct AccountResponse {
+        pub account: AccountInfo,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct AccountInfo {
+        pub account_number: String,
+        pub sequence: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct AccountPrefix {
+        pub bech32_prefix: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct NodeInfo {
+        pub default_node_info: DefaultNodeInfo,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct DefaultNodeInfo {
+        pub network: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct StakingParams {
+        pub params: StakingParamsInner,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct StakingParamsInner {
+        pub bond_denom: String,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct TopicResponse {
+        pub topic: TopicInfo,
+    }
+
+    #[derive(Debug, Deserialize)]
+    pub struct TopicInfo {
+        pub owner: String,
+        pub order: String,
+    }
+
+    #[derive(Clone, Debug, Deserialize)]
+    pub struct SimulateResponse {
+        pub gas_info: GasInfo,
+    }
+
+    #[derive(Clone, Debug, Deserialize)]
+    pub struct GasInfo {
+        pub gas_used: String,
+    }
+
+    #[derive(Clone, Debug, Deserialize)]
+    pub struct Config {
+        pub minimum_gas_price: String,
+    }
+}
+
+#[derive(Copy, Clone, Debug, Default)]
+pub struct AccountMetadata {
+    pub account_number: u64,
+    pub sequence: u64,
+}
+
+#[derive(Clone, Debug, serde::Serialize)]
+pub struct SimulateRequest {
+    pub tx_bytes: String,
+}
+
+pub async fn estimate_transaction(
```

Comment:
do we want any retry logic here? can be a follow up PR

Reply:
The current implementation doesn't care if we are not able to submit the blob. We just *try* to do that, and if not, then it's okay.

In the future(follow-up issue), we need to worry about it and guarantee that all blobs are posted and there are no gaps.